src/console_log.rs

```diff
@@ -71,3 +62,432 @@ impl ConsoleLogHandler {
         log::info!("{}", message.cyan());
     }
 }
+
+/// Returns the well-known log [Function]s.
+fn get_log_functions() -> Vec<Function> {
+    LOG_FUNCTIONS
+        .iter()
+        .map(|func_decl| {
+            let (name, params) = func_decl
+                .trim_end_matches(")")
+                .split_once("(")
+                .unwrap_or_else(|| panic!("unable to obtain function name for '{}'", func_decl));
+
+            #[allow(deprecated)] // for deprecated field `constant`
+            Function {
+                name: String::from(name),
+                inputs: params
+                    .split(",")
+                    .into_iter()
+                    .enumerate()
+                    .map(|(index, param)| Param {
+                        name: format!("p{index}"),
+                        kind: Reader::read(param.clone()).unwrap_or_else(|err| {
+                            panic!(
+                                "failed deserializing type '{}' for '{}' : {:?}",
+                                param, func_decl, err
+                            )
+                        }),
+                        internal_type: Some(String::from(param)),
+                    })
+                    .collect(),
+                outputs: vec![],
+                constant: false,
+                state_mutability: StateMutability::View,
+            }
+        })
+        .collect()
+}
+
+/// Represents the common log functions.
+/// See https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/console.sol
+const LOG_FUNCTIONS: [&str; 378] = [
+    "log(int256)",
```

Comment:
I can also dynamically generate this list, though unsure which is better.

```rust
        let mut func_decl = IndexSet::<String>::default();
        func_decl.extend([
            String::from("log(int256)"),
            String::from("log(uint256)"),
            String::from("log(string)"),
            String::from("log(bool)"),
            String::from("log(address)"),
            String::from("log(bytes)"),
        ]);

        // bytes1 - bytes32
        func_decl.extend(
            (1..=32)
                .into_iter()
                .map(|i| format!("log(bytes{i})"))
                .collect_vec(),
        );

        // combination of ("uint256","string", "bool", "address") upto 4 params
        for i in 1..=4 {
            func_decl.extend(
                ["uint256", "string", "bool", "address"]
                    .iter()
                    .cloned()
                    .combinations_with_replacement(i)
                    .map(|c| {
                        c.into_iter()
                            .permutations(i)
                            .map(|c| format!("log({})", c.join(",")))
                    })
                    .flatten()
                    .collect_vec(),
            );
        }
```

Reply:
I'm fine either way tbh. I'm leaning towards the explicit list as I don't think `Console.sol` is expected to change from Hardhat, and it provides a clear list of what's supported.

Reply:
Yes, I'm also leaning towards a hardcoded list.